add selectored singledispatch, backports_abc and import test for gen

bollwyvl opened the pull request:
Just got this error over on [a build](https://travis-ci.org/Anaconda-Platform/nb_conda/jobs/147756028#L917):

``` python
File "/home/travis/miniconda/envs/nb_conda/lib/python2.7/site-packages/tornado/gen.py", line 98, in <module>

    from singledispatch import singledispatch  # backport

ImportError: No module named singledispatch
```

The [tornado deps](https://github.com/tornadoweb/tornado/blame/master/setup.py#L134) haven't changed in some time, but guess that's the brakes.

This adds the deps with selectors based on the above, and an import test for `tornado.gen` where both of these are imported!

Commit:
add selectored singledispatch, backports_abc and import test for gen

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (`recipe`) and found it was in an excellent condition.

jakirkham commented on recipe/meta.yaml:
```diff
@@ -10,25 +10,28 @@ source:
     - python
     - setuptools
-    - ssl_match_hostname  # [py<35]
+    - ssl_match_hostname   # [py<35]
```
Maybe this should just be Python 2.x?

bollwyvl replied:
Reading the `setup.py`:

``` python
if sys.version_info < (2, 7, 9):
  install_requires.append('backports.ssl_match_hostname')
```

Welp, I don't think we've ever built an older python than that, but `[py2k]` would not be unreasonable, I imagine. Waiting to see if the tests run, as is.

jakirkham replied:
They all seem to have passed.

I know these proceeded your proposal, but if you don't mind grabbing them too that would be nice.

jakirkham commented on recipe/meta.yaml:
```diff
@@ -10,25 +10,28 @@ source:
+    - ssl_match_hostname   # [py<35]
     - certifi
+    - singledispatch       # [py<34]
+    - backports_abc >=0.4  # [py<35]
```
👍

jakirkham commented on recipe/meta.yaml:
```diff
@@ -10,25 +10,28 @@ source:
-    - ssl_match_hostname  # [py<35]
+    - ssl_match_hostname   # [py<35]
     - certifi
+    - singledispatch       # [py<34]
```
👍

jakirkham commented on recipe/meta.yaml:
```diff
@@ -10,25 +10,28 @@ source:
   run:
     - python
-    - ssl_match_hostname  # [py<35]
+    - ssl_match_hostname   # [py<35]
```
Same question as above.

jakirkham commented:
Sorry about that @bollwyvl. The change looks reasonable. Though I'm now questioning some of the stuff that was already here. Do you have any thoughts on it?

Commit:
add import tests for netutil, normalize selectors based on setup.py

bollwyvl commented:
I've updated the deps based on the most literal reading of the `setup.py` I can muster. Also added a test for netutil, which acttually uses the certifi stuff. I think the full test spec is a task for another day, as it includes integration testing for a number of things in a heavy matrix (asyncio, curl, etc.) What else were you looking for?

jakirkham commented on recipe/meta.yaml:
```diff
@@ -10,25 +10,27 @@ source:
-    - ssl_match_hostname  # [py<35]
-    - certifi
+    - ssl_match_hostname   # [py2k]
+    - certifi              # [py2k]
```
Looks like this is Python ~~3.4~~ 3.3 and below.

Edit: Had constraint wrong.

bollwyvl replied:
so,

``` yaml
- ssl_match_hostname   # [py<35]
- certifi   # [py<35] 
```

jakirkham replied:
Sorry Python 3.3 and below.

jakirkham replied:
so,

``` yaml
- singledispatch   # [py<34]
- certifi   # [py<34] 
```

jakirkham commented on recipe/meta.yaml:
```diff
@@ -10,25 +10,27 @@ source:
-    - certifi
+    - ssl_match_hostname   # [py2k]
+    - certifi              # [py2k]
+    - singledispatch       # [py2k]
```
Same.

rsignell-usgs commented:
Earlier this morning I did `conda install jupyter` and found that I needed to install not only `singledispatch` but `backports_abc`.    Is that covered here?

bollwyvl commented:
> Earlier this morning I did conda install jupyter and found that I needed to install not only singledispatch but backports_abc. Is that covered here?

Yup! Thanks for the verification!

jakirkham commented:
Two tweaks and I think this is good to go, @bollwyvl.

jakirkham commented:
Gah! Sorry, will just add a PR to yours for the changes as I feel I'm being confusing.

bollwyvl commented:
> Gah! Sorry, will just add a PR to yours for the changes as I feel I'm being confusing.

Thanks!

jakirkham committed:
recipe: Add ordereddict and tweak missing dependency versions.

jakirkham commented:
Sorry about that. Here's PR ( https://github.com/bollwyvl/tornado-feedstock/pull/1 ) where I tried my best to follow `setup.py` to the letter.

bollwyvl committed:
Merge pull request #1 from jakirkham/update-dependencies_2

Update dependencies (pt. 2)

bollwyvl commented:
Great, merged!

jakirkham commented:
Cool, will merge when passes.

jakirkham commented:
Thanks @bollwyvl.

bollwyvl commented:
Thank you! Back to the extension coal face...

jakirkham commented:
This is now released.

cc @rsignell-usgs
